### ISSUE TITLE
Not working with angular 2.0.0

### ISSUE THREAD

**Issue by DannyGolhar** [afrad/angular2-websocket#40]

**Comment by bomberblue07** [comment]
Can you elaborate?

**Comment by schungx** [comment]
Well, the current angular2-websocket.js is in AMD format.  Angular 2 switched to UMD's, so it needs certain processing if not packed with a bundler.  For example, I do my packaging by hand (poor man's webpack) and I have to manually edit the file to convert the `require` calls into a `System.amdDefine`.

**Comment by bomberblue07** [comment]
It works fine for me, but I use SystemJS. Can you submit a PR with the changes you need?

**Comment by schungx** [comment]
Yes, SystemJS can take AMD modules.  That's why.  Angular 2 has removed dependencies on SystemJS and should now, theoretically, work with any module loader.  That's the reason behind the switch to UMD module format.

I suppose you should simply ask TypeScript to generate UMD modules and be done with it.  That should make sure it works in a wide variety of module loaders.

Or separate different versions, such as `angular2-websocket.amd.min.js`, `angular2-websocket.umd.min.js` etc.

**Comment by bomberblue07** [comment]
I'm not the developer here, @afrad is. So he can step in here and comment on what he's willing to do, or you can submit a PR on what you want done instead of describing what you want done.

**Comment by afrad** [comment]
Didn't want to interrupt the discussion.
@schungx all makes sense to me, could you please submit a pr?

**Comment by schungx** [comment]
Sorry!  Not very familiar with this Git thing...!  Am a Windows kind of person.

Can you simply turn the module format in `tsconfig.json` to `umd`?

**Comment by bomberblue07** [comment]
@schungx You are forgiven for being new to Git. PR = pull request. Essentially, you can fork this repository, make the changes to it that you want to see, and submit a pull request to this repository to merge those changes in. If @afrad is cool with your changes, he'll merge them in.
